node/lib/util/commit.js

```diff
@@ -2222,3 +2222,99 @@ it empty. You can remove the commit entirely with "git meta reset HEAD^".`);
                                        message,
                                        subMessages);
 });
+
+const submoduleHasChangesToCommit = function(status, submoduleName) {
```

Comment:
This doesn't work in the case of --amend with no changes.  Actually, I think git meta just doesn't amend in this case either (which is wrong -- it needs to change the committer and commit date), so maybe it's OK that this is also broken. Probably better not to run the hooks if we're not going to run the commit.

So I think this does not need changes.  I understand that you put the hooks here because you want to run all of the hooks before running any of the commits, so there's not much you can do.